### DISCUSSION TITLE
Feature Request: Quick actions on Dashboard lists / Query Logs

### DISCUSSION THREAD

**Discussion by mocollin** [TechnitiumSoftware/DnsServer#634] [Ideas]
Hi!

I recently discovered Technitium DNS server from a DNS discussion in r/homelab on reddit.  Been running 2 resolvers on my network as my main DNS and loving it.  (Looking forward to the clustering features!)

I'd like to request some additional "quick actions" be added to the Dashboard "Top Lists".  Each entry in the list has a 3-dots menu with Block/Unblock, which is great.  But I would also love to have the ability to:

- Whois - Run a Whois query and display results in a modal
- Show Logs - Would switch to the Query Log tab, with either the Client IP, or the Domain name prepopulated, so you can just hit Query to see the logs.  This would make troubleshooting & filtering faster.
- Run DNS Query - Would switch to the DNS Client tab with the Domain name pre-populated.  This would allow you to see which block lists blocked a query, etc.
- Filter Dashboard Stats - This would redraw the stats on the Dashboard just for that Host, or that Domain name.  Basically a filter for the charts.  Would be nice for filtering the Top Domains and Blocks for an individual host.

For the Query Logs I would like to request:

- Add the actions above to the 3-dots menu on the Query Logs
- Differentiate the Different Response Types by color.  Similar to the color-code in the graph on the dashboard.  Color coding could be a checkbox in the Query builder.
- Display the Round Trip Time as a column in the Query Logs.  A lot of what I do is work to make my DNS responses as fast as possible, and being able to see it in the logs would be helpful.

Thanks again!

**Comment by ShreyasZare** [comment]
Thanks for the request and the compliments. Will get some of them added in couple of next updates. There are a few things though:

> Whois - Run a Whois query and display results in a modal

WhoIs is slightly out of scope and is a bit tricky to implement since the responses are human readable texts so it will require some time to have a WhoIs client that can fetch data reliably. It may take time but will try to get this added.

> Filter Dashboard Stats - This would redraw the stats on the Dashboard just for that Host, or that Domain name. Basically a filter for the charts. Would be nice for filtering the Top Domains and Blocks for an individual host.

The dashboard data is not backed by a database so filtering it or pivoting it based on some parameters is not possible. The only filtering possible is with date and time.

**Comment by johall07** [comment]
What about clicking a point on the graph and being taken to the query log with that date and time already in the filter?

**Reply by ShreyasZare** [reply]
The chart library being used here does not support click events.
